src/internal/Path.php

```diff
@@ -0,0 +1,84 @@
+<?php
+
+declare(strict_types=1);
+
+/**
+ * This file is part of vfsStream.
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace bovigo\vfs\internal;
+
+/**
+ * Helper methods for working with pathes.
+ * 
+ * @internal
+ */
+final class Path
+{
+    /** @var  string */
+    private $dirname;
+    /** @var  string */
+    private $basename;
+
+    private function __construct(string $dirname, string $basename)
+    {
+        $this->dirname = $dirname;
+        $this->basename = $basename;
+    }
+
+    public function dirname(): string
+    {
+        return $this->dirname;
+    }
+
+    public function hasDirname(): bool
+    {
+        return !empty($this->dirname);
+    }
+
+    public function basename(): string
+    {
+        return $this->basename;
+    }
+
+    /**
+     * splits path into its dirname and the basename
+     */
+    public static function split(string $path): self
+    {
+        $lastSlashPos = strrpos($path, '/');
+        if ($lastSlashPos === false) {
+            return new Path('', $path);
+        }
+
+        return new Path(
+            substr($path, 0, $lastSlashPos),
+            substr($path, $lastSlashPos + 1),
```

Comment:
```suggestion
            substr($path, $lastSlashPos + 1)
```
This trailing comma breaks the code on 7.2 (which is what I still have installed on my mac).

Unless we're officially dropping 7.2 support. If so, composer.json should be updated.

Reply:
Nice catch. I think we should drop 7.2, but not in this one. :)